instrumentation/khttp-0.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/khttp/KHttpAdvice.java

```diff
@@ -22,33 +23,36 @@ public static void methodEnter(
       @Advice.Argument(value = 0) String method,
       @Advice.Argument(value = 1) String uri,
       @Advice.Argument(value = 2, readOnly = false) Map<String, String> headers,
-      @Advice.Local("otelSpan") Span span,
+      @Advice.Local("otelContext") Context context,
       @Advice.Local("otelScope") Scope scope,
-      @Advice.Local("otelCallDepth") Depth callDepth) {
-
+      @Advice.Local("otelCallDepth") CallDepth callDepth) {
     callDepth = tracer().getCallDepth();
-    if (callDepth.getAndIncrement() == 0) {
-      span = tracer().startSpan(new RequestWrapper(method, uri, headers));
-      if (span.getSpanContext().isValid()) {
-        headers = asWritable(headers);
-        scope = tracer().startScope(span, headers);
-      }
+    if (callDepth.getAndIncrement() != 0) {
+      return;
     }
+    Context parentContext = currentContext();
+    if (!tracer().shouldStartSpan(parentContext)) {
+      return;
+    }
+
+    context = tracer().startSpan(parentContext, new RequestWrapper(method, uri, headers), headers);
+    headers = asWritable(headers);
```

Comment:
I think this was used to allow for context injection into outgoing headers. Are you sure this still works?

Reply:
oh no, thanks for catching this, fixed